adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -109,24 +109,66 @@ object ADAMContext {
   implicit def setToJavaSet[A](set: Set[A]): java.util.Set[A] = setAsJavaSet(set)
 }
 
+/**
+ * A filter to run on globs/directories that finds all files with a given name.
+ *
+ * @param name The name to search for.
+ */
+private class FileFilter(private val name: String) extends PathFilter {
+
+  /**
+   * @param path Path to evaluate.
+   * @return Returns true if the filename of the path matches the name passed
+   *   to the constructor.
+   */
+  def accept(path: Path): Boolean = {
+    path.getName == name
+  }
+}
+
 import org.bdgenomics.adam.rdd.ADAMContext._
 
 class ADAMContext(@transient val sc: SparkContext) extends Serializable with Logging {
 
-  private[rdd] def loadBamDictionary(filePath: String): SequenceDictionary = {
-    val samHeader = SAMHeaderReader.readSAMHeaderFrom(new Path(filePath), sc.hadoopConfiguration)
-    loadBamDictionary(samHeader)
-  }
-
+  /**
+   * @param samHeader The header to extract a sequence dictionary from.
+   * @return Returns the dictionary converted to an ADAM model.
+   */
   private[rdd] def loadBamDictionary(samHeader: SAMFileHeader): SequenceDictionary = {
     SequenceDictionary(samHeader)
   }
 
+  /**
+   * @param samHeader The header to extract a read group dictionary from.
+   * @return Returns the dictionary converted to an ADAM model.
+   */
   private[rdd] def loadBamReadGroups(samHeader: SAMFileHeader): RecordGroupDictionary = {
     RecordGroupDictionary.fromSAMHeader(samHeader)
   }
 
+  /**
+   * @param filePath The (possibly globbed) filepath to load a VCF from.
+   * @return Returns a tuple of metadata from the VCF header, including the
+   *   sequence dictionary and a list of the samples contained in the VCF.
+   */
   private[rdd] def loadVcfMetadata(filePath: String): (SequenceDictionary, Seq[Sample]) = {
+    // get the paths to all vcfs
+    val files = getFsAndFiles(new Path(filePath))
```

Comment:
Also, similar to the `.bam`/`.sam` logic, thoughts here? I don't want to go with the same logic, since `.vcf`s commonly show up GZIPped/etc, but .tbi files abound...